### DISCUSSION TITLE
Custom data in iPXE script (which gets loaded into initrd)

### DISCUSSION THREAD

**Discussion by guettli** [ipxe/ipxe#1083] [Q&A]
We are working with a provider which has an API to start a custom image via iPXE.

But if you choose iPXE, then cloud-init won't run.

AFAIK then kernel command line is limited to few kilobytes. I guess our cloud-init script will be large for a kernel argument.

I would like to avoid that we need to upload the cloud-init data to some random URL, and then provide this random URL to the kernel. This would work, but we need to run an extra service for that.

My crazy idea to have something like a heredoc:

```
#!ipxe

kernel http://tinycorelinux.net/14.x/x86_64/release/distribution_files/vmlinuz64

initrd http://tinycorelinux.net/14.x/x86_64/release/distribution_files/corepure64.gz

initrd <<EOF
my cloud init file which ~ 50 lines long.
....
EOF /etc/foo mode=0644
```

Is it somehow possible to get data from the iPXE file into the initrd?

**Comment by NiKiZe** [comment]
Would a initrd data:base64... Url be a acceptable solution? (Just an idea)

**Reply by guettli** [reply]
@NiKiZe yes, that would be a solution. We create the iPXE file with a script. So encoding the data can be done easily.

**Reply by guettli** [reply]
@NiKiZe here is my context: Cloud-init:

https://cloudinit.readthedocs.io/en/latest/explanation/kernel-cmdline.html#kernel-cloud-config-url-configuration

I would like to get the cloud-init data to the new machine.

I would like to avoid to set up a http server first.
